bookkeeper-benchmark/src/main/java/org/apache/bookkeeper/benchmark/BenchBookie.java

```diff
@@ -35,6 +35,7 @@
 import org.apache.bookkeeper.client.api.WriteFlag;
 import org.apache.bookkeeper.common.util.OrderedExecutor;
 import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.net.BookieId;
```

Comment:
This is good. I am just wondering the real usecase behind this. Do you expect your bookies to change the ip and nodenames ? We had this requirement earlier when we moved to k8s where ip addresses could change, but we worked around by keeping all clients and bookies in the same k8s cluster where the node names are constant.

Reply:
@jvrao primary use cases are:
- dynamic addresses for Bookies (or simply the ability to change hostname/ip for a bookie)
- support more endpoints and protocols for Bookie (now we are stuck to one single host:port endpoint, and we had to use StartTLS, with BP-38 the Bookie advertises all of the endpoints and thus the client will be able in the future to connect to the preferred one)

see https://github.com/apache/bookkeeper/blob/master/site/bps/BP-41-bookieid.md for more details

Reply:
I recalled that, user from puslar side reported same requirements, they want to change a hostname/ip for a bookie